tests/build/embed_modules.srctree

```diff
@@ -0,0 +1,131 @@
+# mode: run
+# tag: embed
+
+CYTHONIZE -j2 embedded1.pyx embedded2.pyx
+CYTHON --embed --embed-modules embedded1,embedded2 main_module.pyx
+PYTHON build.py main_module main_module.c embedded1.c embedded2.c
+PYTHON run.py main_module
+
+################### build.py ########################
+
+import os
+import subprocess
+import sys
+import sysconfig
+from concurrent.futures import ThreadPoolExecutor
+from functools import partial
+
+def get_config_var(name, default=''):
+    return sysconfig.get_config_var(name) or default
+
+IS_WINDOWS = sys.platform == 'win32'
+
+CC = get_config_var('CC', os.environ.get('CC', 'cl' if IS_WINDOWS else 'cc'))
+INCDIR = sysconfig.get_path('include')
+LIBDIR1 = get_config_var('LIBDIR')
+LIBDIR2 = get_config_var('LIBPL')
+PYLIB = get_config_var('LIBRARY')
+PYLIB_DYN = get_config_var('LDLIBRARY')
+if PYLIB_DYN == PYLIB:
+    # no shared library
+    PYLIB_DYN = ''
+else:
+    # 'lib(XYZ).so' -> XYZ
+    PYLIB_DYN = os.path.splitext(PYLIB_DYN)[0]
+    if PYLIB_DYN.startswith('lib'):
+        PYLIB_DYN = PYLIB_DYN[3:]
+
+CFLAGS = get_config_var('CFLAGS') + ' ' + os.environ.get('CFLAGS', '')
+
+LINKCC = get_config_var('LINKCC', os.environ.get('LINKCC', 'link' if IS_WINDOWS else CC))
+OBJ_EXT = '.obj' if IS_WINDOWS else '.o'
+LINKFORSHARED = get_config_var('LINKFORSHARED')
+LIBS = get_config_var('LIBS')
+SYSLIBS = get_config_var('SYSLIBS')
+EXE_EXT = sysconfig.get_config_var('EXE')
+
+
+def main(args):
+    app_name, *c_files = args
+    c_base_names = [filename[:-2] for filename in c_files]
+    env = os.environ.copy()
+    env['CFLAGS'] = CFLAGS
+    run = partial(subprocess.run, env=env)
+
+    print("CFLAGS:", CFLAGS)
+    print(f"LIBS: {LIBS!r} {SYSLIBS!r} {LINKFORSHARED!r}")
+    print(f"PYLIBS: {PYLIB!r} {PYLIB_DYN!r}")
+
+    compile_commands = [
+        [*CC.split(), '-c', filename+'.c', '-I' + INCDIR]
+        for filename in c_base_names
+    ]
+    for command in compile_commands:
+        print("CC:", command)
+
+    with ThreadPoolExecutor() as pool:
+        for _ in pool.map(run, compile_commands):
+            pass
+
+    linker_command = ([
+        *LINKCC.split(),
+        *[filename + OBJ_EXT for filename in c_base_names],
+        *['-L'+libdir for libdir in [LIBDIR1, LIBDIR2] if libdir],
+        '-l'+PYLIB_DYN if PYLIB_DYN else os.path.join(LIBDIR1, PYLIB),
+        ]
+        + LIBS.split() + SYSLIBS.split() + LINKFORSHARED.split()
+        + ([f'/LIBPATH:{os.path.join(get_config_var("installed_platbase"), "libs")}'] if IS_WINDOWS else [])
+        + ([f'/OUT:{app_name+EXE_EXT}'] if IS_WINDOWS else ['-o', app_name + EXE_EXT])
+    )
+    print("LD:", linker_command)
```

Comment:
```suggestion
    linker_command = [part for part in linker_command if part]
    print("LD:", linker_command)
```

works locally for me.

I think if you have an empty strings in there then it actually looks for a file with no name.

Reply:
So, it can produce an empty string if `PYLIB_DYN`, `LIBDIR1` and `PYLIB` are all empty. I'll fix that instead.

Reply:
What is the right command to run the C compiler on Windows? I found `cl` being mentioned in MS documentation but that doesn't seem to work here.